dvc/external_repo.py

```diff
@@ -19,7 +21,10 @@ def external_repo(url=None, rev=None, rev_lock=None, cache_dir=None):
 
     path = _external_repo(url=url, rev=rev_lock or rev, cache_dir=cache_dir)
     repo = Repo(path)
-    yield repo
+    try:
+        yield repo
+    except NoRemoteError:
+        raise RemoteNotSpecifiedInExternalRepoError(url)
```

Comment:
Just so we can get the traceback if we need it
```suggestion
        raise RemoteNotSpecifiedInExternalRepoError(url, cause=exc)
```